src/main/java/com/minecolonies/structures/lib/TemplateUtils.java

```diff
@@ -0,0 +1,65 @@
+package com.minecolonies.structures.lib;
+
+import com.google.common.base.Functions;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.minecolonies.blockout.Log;
+import com.minecolonies.coremod.blocks.AbstractBlockHut;
+import net.minecraft.init.Blocks;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.gen.structure.template.Template;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+public final class TemplateUtils
+{
+
+    private static final Cache<Template, Map<BlockPos, Template.BlockInfo>> templateBlockInfoCache = CacheBuilder.newBuilder().maximumSize(50).build();
+
+    private TemplateUtils()
+    {
+        throw new IllegalArgumentException("Utils class");
+    }
+
+    public static TileEntity getTileEntityFromPos(final Template template, final BlockPos pos)
+    {
+        final Template.BlockInfo blockInfo = getBlockInfoFromPos(template, pos);
+        if (blockInfo.tileentityData != null)
+        {
+            //TODO: Figure out if this world = null thing does not harm anyone for rendering purposes.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)